plugin/pkg/managedseed/validator/admission.go

```diff
@@ -481,12 +484,15 @@ func (v *ManagedSeed) getSeedDNSProviderForCustomDomain(shoot *gardencore.Shoot)
 		secretRef.Name = *primaryProvider.SecretName
 		secretRef.Namespace = shoot.Namespace
 	} else {
-		secretBinding, err := v.getSecretBinding(shoot.Namespace, shoot.Spec.SecretBindingName)
+		if shoot.Spec.SecretBindingName == nil {
```

Comment:
If it is workerless then the control will not come to getSeedDNSProviderForCustomDomain. It it has come here that means the shoot has workers. In the shoot you have made a lot of fields as pointers but you only have a check for SecretBindingName being nil and if it is you return an error, why?

Reply:
We thought of this, for workerless it won't reach here, yes.
But for shoot with workers also, the check although in static validation (which comes after the admission plugins) is executed.
For a managedseed creation, the referred shoot is already present (and properly validated).
We can remove it as well.